examples/resources/cockroach_egress_private_endpoint/resource.tf

```diff
@@ -0,0 +1,110 @@
+# Example: AWS endpoint for an application service
+resource "cockroach_egress_private_endpoint" "aws_app_service" {
+  cluster_id                = cockroach_cluster.my_aws_cluster.id
+  region                    = "us-east-1"
+  target_service_type       = "PRIVATE_SERVICE"
+  target_service_identifier = "com.amazonaws.vpce.us-east-1.vpce-svc-01234567890abcdef"
+}
+
+# Example: GCP endpoint for an application service
+resource "cockroach_egress_private_endpoint" "gcp_app_service" {
+  cluster_id                = cockroach_cluster.my_gcp_cluster.id
+  region                    = "us-east1"
+  target_service_type       = "PRIVATE_SERVICE"
+  target_service_identifier = "projects/project/regions/us-east1/serviceAttachments/0123456789abcdef"
+}
+
+
+# Example: AWS MSK with SASL/SCRAM authentication
+resource "cockroach_egress_private_endpoint" "msk_sasl_scram" {
+  cluster_id                = cockroach_cluster.my_aws_cluster.id
+  region                    = "us-west-2"
+  target_service_type       = "MSK_SASL_SCRAM"
+  target_service_identifier = "arn:aws:kafka:us-west-2:123456789012:cluster/my-cluster/12345678-1234-1234-1234-123456789012-2"
+}
+
+# Example: AWS MSK with IAM authentication
+resource "cockroach_egress_private_endpoint" "msk_iam" {
+  cluster_id                = cockroach_cluster.my_aws_cluster.id
+  region                    = "us-west-2"
+  target_service_type       = "MSK_SASL_IAM"
+  target_service_identifier = "arn:aws:kafka:us-west-2:123456789012:cluster/my-iam-cluster/87654321-4321-4321-4321-210987654321-3"
+}
+
+# Example: AWS MSK with TLS authentication
+resource "cockroach_egress_private_endpoint" "msk_tls" {
+  cluster_id                = cockroach_cluster.my_aws_cluster.id
+  region                    = "eu-west-1"
+  target_service_type       = "MSK_TLS"
+  target_service_identifier = "arn:aws:kafka:eu-west-1:123456789012:cluster/my-tls-cluster/abcdef12-abcd-abcd-abcd-abcdef123456-4"
+}
```

Comment:
nit: since `us-east-1` is the only region in `my_aws_cluster`, i think we should use that for the examples instead of `us-west-2` and `eu-west-1`